scripts/install-build-tools.sh

```diff
@@ -14,30 +15,216 @@ if (( $EUID != 0 )); then
     SUDO='sudo'
 fi
 
+# install in a custom prefix rather than /usr/local. by default, this
+# chooses "prefix" directory alongside "scripts" directory.
+ROOT=$(cd "$(dirname "$0")"/.. && pwd)
+PREFIX="$(cd "$(dirname "$0")"/.. && pwd)/prefix"
+echo "Will install local dependencies in the following prefix: $PREFIX"
+mkdir -p "$PREFIX"/{bin,lib,include}
+
+# Supporting these versions for buildflow
+PYTHON_VERSIONS=("3.12" "3.11" "3.10")
+ENV_NAME=".py_venv"
+PY_INSTALLED=''
+echo "Python3 versions supported: ${PYTHON_VERSIONS[@]}"
+
+# make a virtual environement to install python packages
+create_venv_install_python() {
+  PY_LOC=$1
+  if [ -z "$PY_LOC" ]; then
+    echo "Python path not provided"
+    exit 1
+  fi
+  PY_VERSION=$2
+  if [ -z "$PY_VERSION" ]; then
+    echo "python version not provided"
+    exit 1
+  fi
+  # remove existing virtual environment if it exists - global var
+  ENV_LOCATION="${ROOT}/${ENV_NAME}"
+  if [ -d "${ENV_LOCATION}" ]; then
+    if rm -rf "${ENV_LOCATION}"; then
+      echo "Removed existing virtual environment"
+    else
+      echo "Failed to remove existing virtual environment"; exit 1
+    fi
+  fi
+  # install pip for linux
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    if ! $SUDO apt install -y python3-pip; then
+      echo "Failed to install python3-pip"
+      wget https://bootstrap.pypa.io/get-pip.py
+      $SUDO python${PY_VERSION} get-pip.py
+      rm get-pip.py
+    fi
+    # install python3 venv module for linux
+    if ! $SUDO apt install -y "python${PY_VERSION}-venv"; then
+      echo "Failed to install python${PY_VERSION}-venv"
+    else
+      echo "Installed python${PY_VERSION}-venv"
+    fi
+  fi
+  # create virtual environment with specific python version
+  if "${PY_LOC}" -m venv "${ENV_NAME}"; then
+    echo "Virtual environment '${ENV_NAME}' created"
+  else
+    echo "Virtual environment creation failed"; exit 1
+  fi
+  # activate virtual environment
```

Comment:
Why not use the `activate-venv.sh` script you included?

Reply:
ACK - added